diff --git a/meson.build b/meson.build
index 8a26674..b5f5285 100644
--- a/meson.build
+++ b/meson.build
@@ -40,32 +40,8 @@ sources = ['src/base64.c']
 libtrower = library(meson.project_name(),
                     sources,
                     include_directories: inc,
-                    install: true)
-
-################################################################################
-# Define the tests
-################################################################################
-
-if not meson.is_subproject()
-  test_args = ['-fprofile-arcs', '-ftest-coverage', '-O0']
-
-  cunit_dep = dependency('cunit')
-
-  test('simple test',
-       executable('simple', ['tests/simple.c'],
-                  include_directories: inc,
-                  dependencies: cunit_dep,
-                  install: false,
-                  link_args: test_args,
-                  link_with: libtrower))
-
-  add_test_setup('valgrind',
-                 is_default: true,
-                 exe_wrapper: [ 'valgrind',
-                                '--leak-check=full',
-                                '--error-exitcode=1' ],
-                 timeout_multiplier: 2)
-endif
+                    install: true,
+                    soversion: '1')
 
 ################################################################################
 # Add the license check target
